clang/lib/AST/Availability.cpp

```diff
@@ -16,33 +16,101 @@
 #include "clang/AST/Decl.h"
 #include "clang/Basic/TargetInfo.h"
 
-namespace clang {
+namespace {
+
+struct AvailabilitySet {
```

Comment:
Can you add a doc string or change the type name to match closer to what this captures? IIUC it's for member attributes' availability